### DISCUSSION TITLE
INCOMPATIBLE_DESTINATION这个错误是怎么引起的呢

### DISCUSSION THREAD

**Discussion by AnBuNengSi** [rts-cn/rts#50] [XSwitch]
`callbacks is null

FreeSwitch2.vue:90 token 019232ca-dc6b-7fdd-8c80-49cf53a67209
FreeSwitch2.vue:180 verto正在连接
FreeSwitch2.vue:181 callbacks is not null

FreeSwitch2.vue:181 options tag webcam
FreeSwitch2.vue:181 verto ws connect to : ws://172.31.146.57:8081/ws
FreeSwitch2.vue:181 CACHED RES FOR CAM any {validRes: Array(0), bestResSupported: Array(2)}
@xswitch_rtc.js?v=a8900d54:8965 verto ws connected
FreeSwitch2.vue:167 当 verto 成功登录之后触发的回调函数   true
FreeSwitch2.vue:103 onMessage {name: 'clientReady', val: 3} {reattached_sessions: Array(0)}
FreeSwitch2.vue:231 refreshDeviceList 刷新设备列啦啦啦
FreeSwitch2.vue:235 MediaDevices is supported
FreeSwitch2.vue:277 enumerating devices
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audioinput 默认 - 麦克风 (720P USB Camera-Audio) (2bdf:0280)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audioinput 通讯 - 麦克风 (720P USB Camera-Audio) (2bdf:0280)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audioinput 麦克风 (720P USB Camera-Audio) (2bdf:0280)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : videoinput 720P USB Camera (2bdf:0280)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audiooutput 默认 - Realtek Digital Output (Realtek High Definition Audio)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audiooutput 通讯 - Realtek Digital Output (Realtek High Definition Audio)
@xswitch_rtc.js?v=a8900d54:9884 devs[i] kind : audiooutput Realtek Digital Output (Realtek High Definition Audio)
@xswitch_rtc.js?v=a8900d54:9839 InputDeviceInfo {deviceId: 'default', kind: 'audioinput', label: '默认 - 麦克风 (720P USB Camera-Audio) (2bdf:0280)', groupId: 'f491eb21789ff2b1cb71bb6f8a427f64cc86c77e9f24e3af5adaf8c9700b3438'}
@xswitch_rtc.js?v=a8900d54:9840 audioinput: 默认 - 麦克风 (720P USB Camera-Audio) (2bdf:0280) id = default
@xswitch_rtc.js?v=a8900d54:9839 InputDeviceInfo {deviceId: 'communications', kind: 'audioinput', label: '通讯 - 麦克风 (720P USB Camera-Audio) (2bdf:0280)', groupId: 'f491eb21789ff2b1cb71bb6f8a427f64cc86c77e9f24e3af5adaf8c9700b3438'}
@xswitch_rtc.js?v=a8900d54:9840 audioinput: 通讯 - 麦克风 (720P USB Camera-Audio) (2bdf:0280) id = communications
@xswitch_rtc.js?v=a8900d54:9839 InputDeviceInfo {deviceId: 'a214e91f660e1ba46d3ed559a0f0d4791fa57869707f1d6768e4ac87452b67a7', kind: 'audioinput', label: '麦克风 (720P USB Camera-Audio) (2bdf:0280)', groupId: 'f491eb21789ff2b1cb71bb6f8a427f64cc86c77e9f24e3af5adaf8c9700b3438'}
@xswitch_rtc.js?v=a8900d54:9840 audioinput: 麦克风 (720P USB Camera-Audio) (2bdf:0280) id = a214e91f660e1ba46d3ed559a0f0d4791fa57869707f1d6768e4ac87452b67a7
@xswitch_rtc.js?v=a8900d54:9839 InputDeviceInfo {deviceId: '7b31c5f17e05177874a07e0d3df3ac8672fd8b1568883ff1255a71da109aa87d', kind: 'videoinput', label: '720P USB Camera (2bdf:0280)', groupId: 'f491eb21789ff2b1cb71bb6f8a427f64cc86c77e9f24e3af5adaf8c9700b3438'}
@xswitch_rtc.js?v=a8900d54:9840 videoinput: 720P USB Camera (2bdf:0280) id = 7b31c5f17e05177874a07e0d3df3ac8672fd8b1568883ff1255a71da109aa87d
@xswitch_rtc.js?v=a8900d54:9839 MediaDeviceInfo {deviceId: 'default', kind: 'audiooutput', label: '默认 - Realtek Digital Output (Realtek High Definition Audio)', groupId: 'd1942f4eb68f80c80d2893702dbc8b6a83d8b7aca0e8f1e94babc6fd3f1f21ae'}
@xswitch_rtc.js?v=a8900d54:9840 audiooutput: 默认 - Realtek Digital Output (Realtek High Definition Audio) id = default
@xswitch_rtc.js?v=a8900d54:9839 MediaDeviceInfo {deviceId: 'communications', kind: 'audiooutput', label: '通讯 - Realtek Digital Output (Realtek High Definition Audio)', groupId: 'd1942f4eb68f80c80d2893702dbc8b6a83d8b7aca0e8f1e94babc6fd3f1f21ae'}
@xswitch_rtc.js?v=a8900d54:9840 audiooutput: 通讯 - Realtek Digital Output (Realtek High Definition Audio) id = communications
@xswitch_rtc.js?v=a8900d54:9839 MediaDeviceInfo {deviceId: 'c3dffb081dc1fbd3bbf8c1e63c5e30b9669d39c862db72859d66712ff677f878', kind: 'audiooutput', label: 'Realtek Digital Output (Realtek High Definition Audio)', groupId: 'd1942f4eb68f80c80d2893702dbc8b6a83d8b7aca0e8f1e94babc6fd3f1f21ae'}
@xswitch_rtc.js?v=a8900d54:9840 audiooutput: Realtek Digital Output (Realtek High Definition Audio) id = c3dffb081dc1fbd3bbf8c1e63c5e30b9669d39c862db72859d66712ff677f878
@xswitch_rtc.js?v=a8900d54:9865 Audio IN Devices (3) [{…}, {…}, {…}]
@xswitch_rtc.js?v=a8900d54:9866 Audio Out Devices (3) [{…}, {…}, {…}]
@xswitch_rtc.js?v=a8900d54:9867 Video Devices [{…}]
FreeSwitch2.vue:240 refreshDevices runtime true
FreeSwitch2.vue:243 [{…}]
FreeSwitch2.vue:244 (3) [{…}, {…}, {…}]
FreeSwitch2.vue:245 (3) [{…}, {…}, {…}]
FreeSwitch2.vue:67 开始呼叫
FreeSwitch2.vue:68 tag id webcam
VertoDialog2 @ @xswitch_rtc.js?v=a8900d54:7027
Verto2.newCall @ @xswitch_rtc.js?v=a8900d54:9758
call @ FreeSwitch2.vue:68
_createElementVNode.onClick._cache.<computed>._cache.<computed> @ FreeSwitch2.vue:14
callWithErrorHandling @ chunk-544CHW4L.js?v=a8900d54:1678
callWithAsyncErrorHandling @ chunk-544CHW4L.js?v=a8900d54:1685
invoker @ chunk-544CHW4L.js?v=a8900d54:10355
Show 5 more frames
Show less
FreeSwitch2.vue:119 beforeCall VertoDialog2 {params: {…}, videoStream: null, verto: Verto2, direction: {…}, lastState: {…}, …}
@xswitch_rtc.js?v=a8900d54:6072 options {useVideo: null, useStereo: false, userData: null, localVideo: video#webcamlocal, screenShare: false, …}
@xswitch_rtc.js?v=a8900d54:6084 Camera Disabled
@xswitch_rtc.js?v=a8900d54:6590 Call Audio constraints {googAutoGainControl: false, googNoiseSuppression: false, googHighpassFilter: false, googEchoCancellation: false, autoGainControl: false, …}
@xswitch_rtc.js?v=a8900d54:6591 Call Video constraints false
@xswitch_rtc.js?v=a8900d54:4610 Stream Success
@xswitch_rtc.js?v=a8900d54:7126 stream started
@xswitch_rtc.js?v=a8900d54:4688 audio first
@xswitch_rtc.js?v=a8900d54:4690 Offer SDP============
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:2379723607 1 udp 2122260223 192.168.31.1…121 typ host generation 0 ufrag iPXX network-id 1', sdpMid: '0', sdpMLineIndex: 0, foundation: '2379723607', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:4170950995 1 udp 2122194687 172.16.0.103…122 typ host generation 0 ufrag iPXX network-id 2', sdpMid: '0', sdpMLineIndex: 0, foundation: '4170950995', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:1799537927 1 udp 2122129151 172.31.146.1…123 typ host generation 0 ufrag iPXX network-id 3', sdpMid: '0', sdpMLineIndex: 0, foundation: '1799537927', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:1140629756 1 udp 2122063615 192.168.1.20…124 typ host generation 0 ufrag iPXX network-id 4', sdpMid: '0', sdpMLineIndex: 0, foundation: '1140629756', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:1791628736 1 udp 2121998079 192.168.30.1…125 typ host generation 0 ufrag iPXX network-id 5', sdpMid: '0', sdpMLineIndex: 0, foundation: '1791628736', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:1937598403 1 tcp 1518280447 192.168.31.1…ptype active generation 0 ufrag iPXX network-id 1', sdpMid: '0', sdpMLineIndex: 0, foundation: '1937598403', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:103906759 1 tcp 1518214911 172.16.0.103 …ptype active generation 0 ufrag iPXX network-id 2', sdpMid: '0', sdpMLineIndex: 0, foundation: '103906759', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:2515014035 1 tcp 1518149375 172.31.146.1…ptype active generation 0 ufrag iPXX network-id 3', sdpMid: '0', sdpMLineIndex: 0, foundation: '2515014035', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:3176559720 1 tcp 1518083839 192.168.1.20…ptype active generation 0 ufrag iPXX network-id 4', sdpMid: '0', sdpMLineIndex: 0, foundation: '3176559720', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:7116 offer RTCIceCandidate {candidate: 'candidate:2489368916 1 tcp 1518018303 192.168.30.1…ptype active generation 0 ufrag iPXX network-id 5', sdpMid: '0', sdpMLineIndex: 0, foundation: '2489368916', component: 'rtp', …}
@xswitch_rtc.js?v=a8900d54:4628 ICE Complete
@xswitch_rtc.js?v=a8900d54:4637 ICE SDP
@xswitch_rtc.js?v=a8900d54:7077 RECV offer SDP v=0
o=- 3130908245398091770 2 IN IP4 127.0.0.1
s=-
t=0 0
a=group:BUNDLE 0
a=extmap-allow-mixed
a=msid-semantic: WMS 7b029c25-4fde-4df7-8344-2a4761a9ecec
m=audio 64121 UDP/TLS/RTP/SAVPF 111 63 9 0 8 13 110 126
c=IN IP4 192.168.31.1
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:2379723607 1 udp 2122260223 192.168.31.1 64121 typ host generation 0 network-id 1
a=candidate:4170950995 1 udp 2122194687 172.16.0.103 64122 typ host generation 0 network-id 2
a=candidate:1799537927 1 udp 2122129151 172.31.146.103 64123 typ host generation 0 network-id 3
a=candidate:1140629756 1 udp 2122063615 192.168.1.203 64124 typ host generation 0 network-id 4
a=candidate:1791628736 1 udp 2121998079 192.168.30.1 64125 typ host generation 0 network-id 5
a=candidate:1937598403 1 tcp 1518280447 192.168.31.1 9 typ host tcptype active generation 0 network-id 1
a=candidate:103906759 1 tcp 1518214911 172.16.0.103 9 typ host tcptype active generation 0 network-id 2
a=candidate:2515014035 1 tcp 1518149375 172.31.146.103 9 typ host tcptype active generation 0 network-id 3
a=candidate:3176559720 1 tcp 1518083839 192.168.1.203 9 typ host tcptype active generation 0 network-id 4
a=candidate:2489368916 1 tcp 1518018303 192.168.30.1 9 typ host tcptype active generation 0 network-id 5
a=ice-ufrag:iPXX
a=ice-pwd:jAQ0Qrjm/lxa3leSHMmyeZrZ
a=ice-options:trickle
a=fingerprint:sha-256 C0:F4:74:71:A3:4A:E8:CC:B8:5B:13:5C:63:61:3C:5B:8E:DE:F7:42:41:67:87:B7:59:B8:60:A0:AE:88:BE:4A
a=setup:actpass
a=mid:0
a=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level
a=extmap:2 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time
a=extmap:3 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01
a=extmap:4 urn:ietf:params:rtp-hdrext:sdes:mid
a=sendrecv
a=msid:7b029c25-4fde-4df7-8344-2a4761a9ecec 388504b4-9025-4227-9596-4cf22ea23cc9
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:111 opus/48000/2
a=rtcp-fb:111 transport-cc
a=fmtp:111 minptime=10;useinbandfec=1
a=rtpmap:63 red/48000/2
a=fmtp:63 111/111
a=rtpmap:9 G722/8000
a=rtpmap:0 PCMU/8000
a=rtpmap:8 PCMA/8000
a=rtpmap:13 CN/8000
a=rtpmap:110 telephone-event/48000
a=rtpmap:126 telephone-event/8000
a=ssrc:2929718238 cname:UN7yAf5toq6KMrxI
a=ssrc:2929718238 msid:7b029c25-4fde-4df7-8344-2a4761a9ecec 388504b4-9025-4227-9596-4cf22ea23cc9

@xswitch_rtc.js?v=a8900d54:7374 Dialog eb3d4d86-213f-8311-502b-358f57e0f90b: state change from new to requesting
@xswitch_rtc.js?v=a8900d54:7377 dialog.state:[object Object]
@xswitch_rtc.js?v=a8900d54:7433 Response: verto.invite State:requesting true {message: 'CALL CREATED', callID: 'eb3d4d86-213f-8311-502b-358f57e0f90b', sessid: '0b02510e-ce3a-1714-c88d-c160bdea0684'}
@xswitch_rtc.js?v=a8900d54:7374 Dialog eb3d4d86-213f-8311-502b-358f57e0f90b: state change from requesting to trying
@xswitch_rtc.js?v=a8900d54:7377 dialog.state:[object Object]
FreeSwitch2.vue:134 主叫正在呼叫
@xswitch_rtc.js?v=a8900d54:7540 eb3d4d86-213f-8311-502b-358f57e0f90b hangup cause INCOMPATIBLE_DESTINATION

@xswitch_rtc.js?v=a8900d54:7542 eb3d4d86-213f-8311-502b-358f57e0f90b set hangup state

@xswitch_rtc.js?v=a8900d54:7374 Dialog eb3d4d86-213f-8311-502b-358f57e0f90b: state change from trying to hangup
@xswitch_rtc.js?v=a8900d54:7377 dialog.state:[object Object]
FreeSwitch2.vue:146 通话挂断
@xswitch_rtc.js?v=a8900d54:7410 verto bye>>>>>
@xswitch_rtc.js?v=a8900d54:7374 Dialog eb3d4d86-213f-8311-502b-358f57e0f90b: state change from hangup to destroy
@xswitch_rtc.js?v=a8900d54:7377 dialog.state:[object Object]
FreeSwitch2.vue:150 通话销毁
@xswitch_rtc.js?v=a8900d54:5875 stopping peer
@xswitch_rtc.js?v=a8900d54:7433 Response: verto.bye State:destroy false {callID: 'eb3d4d86-213f-8311-502b-358f57e0f90b', message: 'CALL DOES NOT EXIST', code: -32002}callID: "eb3d4d86-213f-8311-502b-358f57e0f90b"code: -32002message: "CALL DOES NOT EXIST"[[Prototype]]: Object
@xswitch_rtc.js?v=a8900d54:7540 eb3d4d86-213f-8311-502b-358f57e0f90b hangup cause INCOMPATIBLE_DESTINATION`

以上是浏览器F12的调试日志。这是我的调用过程，通过PC上注册Verto 拨打sip话机，后台web管理也看不到我pc注册成功，前端pc参照webrtc Verto调用去做的，用的vue构建。错误返回是INCOMPATIBLE_DESTINATION

Verto登录参数是 
`{
    "login": "1004@172.31.146.57:7060",
    "passwd": "Xswitch1234",
    "socketUrl": "ws://172.31.146.57:8081/ws",
    "loginParams": {
        "xui_sessid": "019232ca-dc6b-7fdd-8c80-49cf53a67209"
    },
    "localTag": "webcamlocal",
    "useVideo": false,
    "tag": "webcam",
    "ringer_tag": "ringer",
    "ringFile": "/src/components/sounds/bell_ring2.wav",
    "audioParams": {
        "googAutoGainControl": false,
        "googNoiseSuppression": false,
        "googHighpassFilter": false,
        "googEchoCancellation": false,
        "autoGainControl": false,
        "noiseSuppression": false,
        "highpassFilter": false,
        "echoCancellation": false
    },
    "videoParams": {
        "minWidth": 1280,
        "maxWidth": 1280,
        "minHeight": 720,
        "maxHeight": 720,
        "minFramerate": 15
    },
    "deviceParams": {
        "useCamera": "any",
        "useMic": "any",
        "useSpeak": "any"
    },
    "iceServers": false
}`

**Comment by yline** [comment]
我也遇到了类似的问题，有结果了么？

**Comment by seven1240** [comment]
参数 sip profile 里那些 candidate acl 可能要加到  verto 里去。
